front/app/containers/Admin/projects/project/projectHeader/ShareLink/index.tsx

```diff
@@ -0,0 +1,180 @@
+    navigator.clipboard.writeText(
+      `${appConfiguration?.data.attributes.host}/${link}`
+    );
+    setLinkIsCopied(true);
+    trackEventByName(tracks.copyProjectPreviewLink);
+  };
+
+  const { formatMessage } = useIntl();
+
+  const refreshLink = () => {
+    refreshProjectPreviewToken(
+      { projectId },
+      {
+        onSuccess: () => {
+          setLinkIsCopied(false);
+          setRefreshModalIsOpen(false);
+          trackEventByName(tracks.regenerateProjectPreviewLink);
+        },
+      }
+    );
+  };
+  return (
+    <Box position="relative">
+      <Button
+        id="e2e-share-link"
+        buttonStyle="secondary-outlined"
+        icon="share"
+        size="s"
+        padding="4px 8px"
+        onClick={() => {
+          setShareDropdownIsOpen(!shareDropdownIsOpen);
+          setLinkIsCopied(false);
+        }}
+      >
+        {formatMessage(messages.share)}
+      </Button>
+      <Dropdown
+        width="600px"
+        mobileWidth="300px"
+        zIndex="100000"
+        top="40px"
+        right="0px"
+        opened={shareDropdownIsOpen}
+        onClickOutside={() => {
+          setLinkIsCopied(false);
+        }}
+        content={
+          <Box p="8px">
+            <Box display="flex" justifyContent="space-between">
+              <Text variant="bodyL" my="8px">
+                {formatMessage(messages.shareTitle)}
+              </Text>
+              <Box display="flex" alignItems="center">
+                <Tooltip
+                  content={
+                    isPreviewLinkEnabled
+                      ? ''
+                      : formatMessage(messages.shareLinkUpsellTooltip)
+                  }
+                  disabled={isPreviewLinkEnabled}
+                >
+                  <Button
+                    id="e2e-copy-link"
+                    buttonStyle="text"
+                    padding="8px"
+                    icon={linkIsCopied ? 'check-circle' : 'link'}
+                    iconColor={linkIsCopied ? colors.success : colors.teal500}
+                    iconHoverColor={
+                      linkIsCopied ? colors.success : colors.teal700
+                    }
+                    textColor={colors.teal500}
+                    textHoverColor={colors.teal700}
+                    size="s"
+                    onClick={handleCopyLink}
+                    disabled={!isPreviewLinkEnabled}
+                  >
+                    {linkIsCopied
+                      ? formatMessage(messages.shareLinkCopied)
+                      : formatMessage(messages.shareLink)}
+                  </Button>
+                </Tooltip>
+
+                {isPreviewLinkEnabled && (
+                  <Box display="none" id="e2e-link">
+                    {link}
+                  </Box>
+                )}
+
+                <Tooltip
+                  content={formatMessage(messages.refreshLinkTooltip)}
+                  placement="bottom"
+                >
```

Comment:
Shall we disable this tooltip when the feature is disabled?

Reply:
I don't think it's needed. The button is disabled to it cannot be clicked. Having the explanation there does no harm imo.